package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,240 @@
+
+        """
+        self._setup_frames(self._u.trajectory, start, stop, step)
+        self.start = start
+        self.stop = stop
+        self.step = step
+
+        self.n_components = n_components
+        self.run()
+        self._calculated = True
+
+        if n_components is None:
+            return self.cumulated_variance, self.p_components
+        else:
+            return (self.cumulated_variance[:n_components],
+                    self.p_components[:n_components])
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        self.mean = np.zeros(n_dim,)
+        for i, ts in enumerate(self._u.trajectory[self.start:self.stop:self.step]):
+            x = self._atoms.positions.ravel()
+            self.mean += x
+        self.mean /= self.n_frames
+
+    def _single_frame(self):
+        x = self._atoms.positions.ravel()
+        x -= self.mean
+        self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)
+
+    def _conclude(self):
+        self.cov /= self.n_frames - 1
+        e_vals, e_vects = np.linalg.eig(self.cov)
+        sort_idx = np.argsort(e_vals)[::-1]
+        self.variance = e_vals[sort_idx]
+        self.p_components = e_vects[sort_idx]
+        self.cumulated_variance = (np.cumsum(self.variance) /
+                                   np.sum(self.variance))
+
+    def transform(self, atomgroup=None, n_components=None):
+        """Apply the dimensionality reduction on a trajectory
+
+        Parameters
+        ----------
+        n_components: int, optional
+            The number of components to be projected onto, default maps onto
+            all components.
+        Returns
+        -------
+        pca_space : array, shape (number of frames, number of components)
+        """
+        if atomgroup is not None:
+            atoms = atomgroup
+        else:
+            atoms = self._atoms
+
+        if self._atoms != atoms:
+            warnings.warn('This is a transform for different atom types.')
+
+        xyz = np.array([atoms.positions.copy() for ts in self._u.trajectory])
+        xyz = xyz.reshape(self._u.trajectory.n_frames,
+                          self._n_atoms*3, order='F')
+        if n_components is None:
+            return np.dot(xyz, self.p_components.T)
+        else:
+            return np.dot(xyz, self.p_components[:n_components].T)
+
+
+    def inverse_tranform(self, pca_space):
+        """ Transform PCA-transformed data back to original configuration space.
+
+        Parameters
+        ----------
+        pca_space : array
+            The space corresponding to the trajectory coordinates after the PCA
+            transformation. Assumes current principal components were the
+            components projected onto to create the pca_space.
+
+        Returns
+        -------
+        original_traj : array, shape (number of frames, number of atoms*3)
+        """
+        inv = np.linalg.inv(self.p_components)
```

Comment:
This currently only work is `pca_space` contains all PCA components. It won't work for sub-selection of components. So lets say I only look at the first 2 components and then I want to know how a new extrapolated point looks in the configuration space. I would like to use this method. I think you have to use the pseudo inverse for this.